### PR TITLE
IOC Reputation Command Args isArray Validation

### PR DESCRIPTION
**Pull request by avidan-H**
## Status
Ready

## Related Issues
fixes: https://github.com/demisto/etc/issues/19803

## Description
IOC reputation commands will be required to support batched indicators. Therefore implement validations check that args are marked as `isArray`.

## Required version of Demisto
any

## Does it break backward compatibility?
   - No

## Must have
- [ ] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

## Dependencies
Mention the dependencies of the entity you changed as given from the precommit hooks in checkboxes, and tick after tested them.
- [ ] Dependency 1
- [ ] Dependency 2
- [ ] Dependency 3

## Additional changes
Describe additional changes done, for example adding a function to common server.

## Technical writer review
- [ ] [docs/code_conventions/README.md](https://github.com/demisto/content/blob/91ff9002599aafda465cebf820827f6e2364100c/docs/code_conventions/README.MD)
- [ ] [docs/yaml-file-integration/README.md](https://github.com/demisto/content/edit/ioc-args-isarray-validation/docs/yaml-file-integration/README.MD?pr=%2Fdemisto%2Fcontent%2Fpull%2F4738)

### CODE REVIEW

File: Tests/scripts/hook_validations/integration.py [hunk 1]
```diff
@@ -152,6 +153,27 @@ def is_default_arguments(self):
...
         return self._is_valid
 
+    def is_isarray_arguments(self):
+        """Check if a reputation command's (domain/email/file/ip/url)
+            argument of the same name has the 'isArray' attribute set to True
+
+        Returns:
+            bool. Whether 'isArray' is True
+        """
+        reputation_commands = ['file', 'email', 'domain', 'url', 'ip']
```
Comment (Itay4) [inline Tests/scripts/hook_validations/integration.py]: lets make this a const

File: docs/code_conventions/README.MD [hunk 2]
```diff
@@ -203,6 +203,13 @@ def main():
         return_error(f'Failed to execute {demisto.command()} command. Error: {str(e)}')
 ```
 
+
+## IOC Reputation Commands
+There are two implementation requirements for reputation commands (aka `!file`, `!email`, `!domain`, `!url`, and `!ip`) that are enforced by checks in the [hook_validations](../../Tests/scripts/hook_validations/integration.py).
+- The reputation command's argument of the same name must have `default` set to `True`.
+- The reputation command's argument of the same name must have `isArray` set to `True`.
+
```
Comment (Itay4) [inline docs/code_conventions/README.MD]: lets add a link here to the doc that explains what default and isArray are

File: Tests/scripts/hook_validations/integration.py (hunk 1 above)
Comment (avidan-H) [reply Tests/scripts/hook_validations/integration.py]: I'm going to make this a class variable. cool?

Comment (avidan-H) [reply Tests/scripts/hook_validations/integration.py]: switched to constant

**Comment by Itay4** [comment]
@avidan-H lets resolve conflicts and merge

**Pull request merged**